addons/packages/pinniped/0.4.4/bundle/config/values.yaml

```diff
@@ -1,6 +1,7 @@
 #@data/values
 #@overlay/match-child-defaults missing_ok=True
 ---
+tanzuKubernetesRelease: null
```

Comment:
where does this get set? is it in the addon secret ytt template over in tanzu-framework?

it is kinda a bummer that we need to add yet another system-specific data value - the list of data values is already growing too large for a user to keep track of what they need to set. but i think i understand the need.

the new data value is also one of those that should always be provided by the underlying cluster (similar to `dex.dns.vsphere.*` or `pinniped.supervisor_ca_bundle_data`). i think this is confusing for users because they don't know which data values that they should set and which data values that the addon secret ytt template will provide. again, i understand the need for the data value here, but it seems like it hurts the UX.

i wonder if it would be more helpful to make data values like these more abstract in the future. for example, instead of `tanzuKubernetesRelease`, we could have a `nodeSelector` map data value where users can provide `nodeSelector` map values for their `Deployment`'s. then, the new data value is abstract enough that 1) it can be used to resolve this use case of needing to make sure deployments are scheduled on the right nodes and 2) it could be used by the user for other scheduling constraints outside of reason 1. just food for thought at this point.